staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -357,9 +426,23 @@ func (wc *watchChan) transform(e *event) (res *watch.Event) {
 		}
 		object := wc.watcher.newFunc()
 		if err := wc.watcher.versioner.UpdateObject(object, uint64(e.rev)); err != nil {
-			klog.Errorf("failed to propagate object version: %v", err)
+			utilruntime.HandleError(fmt.Errorf("failed to propagate object version: %v", err))
 			return nil
 		}
+		if !wc.isInitialEventsEndBookmarkSent() {
+			objMeta, err := meta.Accessor(object)
```

Comment:
Can you refactor this part to a common helper function with:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/cacher/cache_watcher.go#L368

in https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/util.go

[in that same refactor PR I asked above]